### ISSUE TITLE
Support pip 8 hashes

### ISSUE THREAD

**Issue by dstufft** [kennethreitz/pip-pop#8]
pip 8 introduces a new feature that allows people to bake hashes into their requirements.txt, unfortunately this breaks ``pip-diff``. An example would be https://github.com/pypa/warehouse/blob/master/requirements/main.txt

Output looks like:

```
Traceback (most recent call last):
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip-grep", line 81, in <module>
    main()
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip-grep", line 76, in main
    grep(**kwargs)
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip-grep", line 45, in grep
    r = Requirements(reqfile)
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip-grep", line 25, in __init__
    self.load(reqfile)
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip-grep", line 36, in load
    for requirement in parse_requirements(reqfile, finder=finder, session=requests):
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip/req/req_file.py", line 89, in parse_requirements
    for req in req_iter:
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip/req/req_file.py", line 172, in process_line
    for req in parser:
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip/req/req_file.py", line 89, in parse_requirements
    for req in req_iter:
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip/req/req_file.py", line 119, in process_line
    opts, _ = parser.parse_args(shlex.split(options_str), defaults)
  File "/app/.heroku/python/lib/python3.5/optparse.py", line 1388, in parse_args
    self.error(str(err))
  File "/app/.heroku/python/lib/python3.5/optparse.py", line 1568, in error
    self.exit(2, "%s: error: %s\n" % (self.get_prog_name(), msg))
  File "/app/tmp/buildpacks/python/vendor/pip-pop/pip/req/req_file.py", line 238, in parser_exit
    raise RequirementsFileParseError(msg)
pip.exceptions.RequirementsFileParseError: pip-grep: error: no such option: --hash
```

**Comment by kennethreitz** [comment]
Thanks for the report — I find it interesting that no Heroku users have reported this yet. I suspect that while many people are excited that `--hash`` is available, almost none are actually using it.

**Comment by dstufft** [comment]
I am a Heroku user ;)

**Comment by kennethreitz** [comment]
@dstufft so what exactly is happening here? is `parse_requirements` bailing when it encounters a `--hash`?

**Comment by dstufft** [comment]
Yes.

**Comment by kennethreitz** [comment]
Is there another function available that doesn't fail in this situation, or perhaps a pre-processing step? What does pip do?

**Comment by dstufft** [comment]
Pip added support for that flag, though actually looking at the code here maybe the real issue is just that ``buildpacks/python/vendor/pip-pop/pip`` needs updated to the lastest pip?

**Comment by kennethreitz** [comment]
Ah, I see. Forgot about that. 

That vendored pip was added to work around another issue. :)

**Comment by edmorley** [comment]
> Thanks for the report — I find it interesting that no Heroku users have reported this yet. I suspect that while many people are excited that `--hash` is available, almost none are actually using it.

The compile actually succeeds even with these exceptions, which is probably why people haven't reported it (https://github.com/heroku/heroku-buildpack-python/issues/302#issuecomment-219692444).

> That vendored pip was added to work around another issue. :)

I've filed #10 for that issue (presuming there wasn't something else as well).

**Comment by kennethreitz** [comment]
The master branch of this buildpack has this functionality restored.

**Comment by edmorley** [comment]
This was fixed in heroku/heroku-buildpack-python#308 (which is currently the upstream for development) and then synced back to this repo in 4f5ebcd - so this issue can now be closed :-)
